File: 800 - rated/LineTrip.java
import java.util.Scanner ;

public class LineTrip{
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in) ;
        int t = sc.nextInt() ;
        while (t-- != 0) {
            int n = sc.nextInt() ;
            int x = sc.nextInt() ;
            int[] array = new int[n] ;
            int minCapacity = Integer.MIN_VALUE ;
            for (int i=0 ; i < n ; i++){
                array[i] = sc.nextInt() ;
                if( i > 0) 
                    minCapacity = Math.max(minCapacity,array[i] - array[i-1]) ;
                }
            minCapacity = Math.max(minCapacity, array[0]) ;
            minCapacity = Math.max(minCapacity,(x - array[n - 1]) * 2 ) ;
            System.out.println(minCapacity) ;
            
            
        }
    }
}

File: 800 - rated/line_trip.py
t = int(input()) 
while t :
    n , x = map(int , input().split())
    array = list(map(int,input().split()))
    minCapacity = array[0] 
    for i in range(1,n) :
        minCapacity = max(minCapacity, array[i]-array[i-1])
    minCapacity = max(minCapacity, (x - array[-1])*2) 
    print(minCapacity)
 
